addons/project_issue/models/project_issue.py

```diff
@@ -0,0 +1,336 @@
+    color = fields.Integer('Color Index')
+    user_email = fields.Char(related='user_id.email', string='User Email', readonly=True)
+    date_action_last = fields.Datetime(string='Last Action', readonly=True)
+    date_action_next = fields.Datetime(string='Next Action', readonly=True)
+    legend_blocked = fields.Char(related="stage_id.legend_blocked", string='Kanban Blocked Explanation')
+    legend_done = fields.Char(related="stage_id.legend_done", string='Kanban Valid Explanation')
+    legend_normal = fields.Char(related="stage_id.legend_normal", string='Kanban Ongoing Explanation')
+
+    @api.multi
+    def _read_group_stage_ids(self, domain, read_group_order=None, access_rights_uid=None):
+        access_rights_uid = access_rights_uid or self.env.uid
+        ProjectTaskType = self.env['project.task.type']
+        order = ProjectTaskType._order
+        if read_group_order == 'stage_id desc':
+            order = "%s desc" % order
+        search_domain = []
+        if 'default_project_id' in self.env.context:
+            search_domain = ['|', ('project_ids', '=', self.env.context['default_project_id']), ('id', 'in', self.ids)]
+        else:
+            search_domain = [('id', 'in', self.ids)]
+        project_task_types = ProjectTaskType.sudo(access_rights_uid).search(search_domain, order=order)
+        result = project_task_types.sudo(access_rights_uid).name_get()
+        fold = {project_task_type.id: project_task_type.fold for project_task_type in project_task_types}
+        return result, fold
+
+    _group_by_full = {
+        'stage_id': _read_group_stage_ids
+    }
+
+    @api.depends('create_date', 'date_closed', 'date_open')
+    def _compute_day(self):
+        Calendar = self.env['resource.calendar']
+        for issue in self:
+            calendar_id = issue.project_id.resource_calendar_id.id or None
+
+            dt_create_date = fields.Datetime.from_string(issue.create_date)
+            if issue.date_open:
+                dt_date_open = fields.Datetime.from_string(issue.date_open)
+                issue.day_open = (dt_date_open - dt_create_date).total_seconds() / (24.0 * 3600)
+                issue.working_hours_open = Calendar._model.get_working_hours(self.env.cr, self.env.uid, calendar_id, dt_create_date, dt_date_open,
+                                                                             compute_leaves=True, resource_id=False, default_interval=(8, 16),
+                                                                             context=self.env.context)
+
+            if issue.date_closed:
+                dt_date_closed = fields.Datetime.from_string(issue.date_closed)
+                issue.day_close = (dt_date_closed - dt_create_date).total_seconds() / (24.0 * 3600)
+                issue.working_hours_close = Calendar._model.get_working_hours(self.env.cr, self.env.uid, calendar_id, dt_create_date, dt_date_closed,
```

Comment:
LINE 91, ```calendar_id``` may be ```None```, in new API method will not called for ```None``` and we need to calculate hours even if ```calendar_id``` is ```None```.